resources/web/docs_js/utils.js

```diff
@@ -139,3 +139,103 @@ export const getCurlText = ({consoleText,
 
   return curlText;
 }
+
+/*
+  Parse query strings in a battle-tested, good practice manner.
+  An extraction of the deparam method from Ben Alman's jQuery BBQ plugin
+  http://benalman.com/projects/jquery-bbq-plugin/
+  Exact source: https://github.com/cowboy/jquery-bbq/blob/8e0064ba68a34bcd805e15499cb45de3f4cc398d/jquery.ba-bbq.js#L466
+  Unit tests maintained separately: http://benalman.com/code/projects/jquery-bbq/unit/
+  Licence: dual-licensed under MIT and GPL, we use https://github.com/cowboy/jquery-bbq/blob/8e0064ba68a34bcd805e15499cb45de3f4cc398d/LICENSE-MIT
+*/
+export function deparam(params, coerce) {
```

Comment:
I don't know the right thing, but I know we have Redux and React in the project. Is there a good way to get them to do this?

Reply:
Apparently not. I did some digging and [they decided to explicitly not process querystr](https://github.com/ReactTraining/react-router/issues/4410) in Router v4 (and later), where this kind of functionality would naturally fall. Instead they allow for the use of any 3rd party lib.

I'm not sure how Preact v8 interacts with this but have been unable to find a Preact solution to it. A project (providing routing among other functionality) uses our exact version of Preact in its docs examples and explicitly states a 3rd party library is required for querystr parsing with that setup: http://meiosis.js.org/docs/routing.html (search "Again, there are several options"). It appears Preact doesn't do it either.

When searching for solutions around Redux it's pretty clear people expect you to use React Router to pass querystr data to Redux for state mgmt.

I've narrowed down the options:

1. One of the 4 libraries from [this list](http://meiosis.js.org/docs/routing.html). From these https://github.com/sindresorhus/query-string seems most popular on NPM and well maintained. It will parse querystr and hash.

2. Use the new-ish [universal web API URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams). No IE support at all, so requires [a seemingly popular and well maintained polyfill](https://github.com/jerrybendy/url-search-params-polyfill).

3. The deparam function I used before I found the above. Upon further digging into solutions around react, this approach has been superseded and should be replaced by 1. or 2. 

I believe 2. is a good choice resulting in the least amount of extra code to maintain and using a fairly well established browser API. The polyfill can be dropped whenever we drop IE support, much as any of the other polyfills we use, and leave us with no custom querystr parsing code at all. My previous implementation had a further problem, so I replaced it with this option. Happy to change it though, naturally.

Reply:
I guess I'd go with either 1 or 2, adding it to `package.json` to get it in. I *think* the polyfill approach is a little better, but I super don't have strong opinions.

Reply:
I did end up [adding the polyfill to package.json and yarn.lock](https://github.com/elastic/docs/pull/1764/commits/fc9e2a9d6e5f5aa291426aeeda654e017a97be14) and [invoking the web API (polyfill in IE)](https://github.com/elastic/docs/pull/1764/commits/6fac8be75a605ff1685df03bceba7f1acb0db16c).

I have also tested this in IE9-IE11 on browserstack.com on Win 7 - Win 10. All work.

Does the PR look ok to you now code-wise?